pkg/block/factory/build.go

```diff
@@ -90,7 +96,14 @@ func BuildS3Client(params *aws.Config, skipVerifyCertificateTestOnly bool) (*ses
 		}
 		client = &http.Client{Transport: tr}
 	}
-	sess, err := session.NewSession(params, aws.NewConfig().WithHTTPClient(client))
+	opts := session.Options{}
+	opts.Config.MergeIn(params, aws.NewConfig().WithHTTPClient(client))
+	opts.CredentialsProviderOptions = &session.CredentialsProviderOptions{
+		WebIdentityRoleProviderOptions: func(wirp *stscreds.WebIdentityRoleProvider) {
+			wirp.ExpiryWindow = renewWebIdentity
```

Comment:
> If I ask for too much, it doesn't trim it, instead I get an error. So I don't want to do that. Now I can probably set it to 1 hour, because that's the minimum possible.

This is why I mentioned checking the [max_duration](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use.html#id_roles_use_view-role-max-session). I also think that making this configurable but default of `1h` is pretty safe, then if you have an error when assuming the role you'll see why and then can change the value. 

> What would I get in return? URLs would be valid for longer. That would reduce the occurrence of bad things due to bugs in the client asking lakeFS for a pre-signed URL, but they would still be there.

Agree that it doesn't solve the problem entirely but for sure makes it better, we might of never even notice it.  

That's a pretty standard practice to set session duration when assuming a role, I don't see why not make it configurable if not dynamic and set 1h. 

WDYT?

Reply:
I made it configurable, but with no default value.  We can always set a value later.  I don't want to break more than I have to for this fix.